src/main/java/com/study/realworld/service/UserService.java

```diff
@@ -0,0 +1,50 @@
+package com.study.realworld.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.JsonObject;
+import com.study.realworld.bean.User;
+import com.study.realworld.common.Errors;
+import com.study.realworld.common.Func;
+import com.study.realworld.dao.UserDao;
+import org.springframework.stereotype.Repository;
+import org.springframework.stereotype.Service;
+
+@Service
+@Repository
+public class UserService {
+    private final UserDao userDao;
+
+    public UserService(UserDao userDao) {
+        this.userDao = userDao;
+    }
+
+    public String users(JsonObject user) throws JsonProcessingException {
+        if (userDao.checkSameName(getString(user, "username"))) {  //닉네임 체크
+            return Func.getErrorJson(Errors.SAME_NICKNAME);
+        } else if (userDao.checkSameEmail(getString(user, "email"))) { //email 체크
+            return Func.getErrorJson(Errors.SAME_EMAIL);
+        }
+
+        if ( userDao.registUser(getString(user, "username"),
+                getString(user, "email"),
+                getString(user, "password")) < 0 ) {
+            //등록하다가 에러났을경우
+            return Func.getErrorJson(Errors.DB);
+        }
+
+        User users = userDao.getUsers(getString(user, "email"));
```

Comment:
```suggestion
        User user = userDao.getUser(getString(user, "email"));
```
단건만 가져오니 위에처럼 바꾸는 것이 좋아보이네요

Reply:
해당 함수는 회원가입단계에서 필요하지 않다 판단하여 삭제하였습니다.
다음에 같은 기능을 넣게 된다면 고려하겠습니다 감사합니다